src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1123,6 +1123,10 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                     scriptConfig.Functions.Add((string)function);
                 }
             }
+            else
+            {
+                scriptConfig.Functions = null;
```

Comment:
No, it's expected to be null (there's a test already that verifies the the initial state).